vm/src/stdlib/itertools.rs

```diff
@@ -1067,7 +1067,7 @@ mod decl {
             PyItertoolsProduct {
                 pools,
                 idxs: PyRwLock::new(vec![0; l]),
-                cur: AtomicCell::new(l - 1),
+                cur: AtomicCell::new(l.saturating_sub(1)),
```

Comment:
Does `wrapping_sub` make any problem here? otherwise i prefer to use wrapping_sub here.
Because we are using `fetch_sub` below, which implies `wrapping_sub`, we just can go the same way.

Reply:
Isn't a wrapping subtraction what was causing the issue in the first place?

Reply:
It does, but this PR also added length check which prevent to hit self.cur when it is overflowed.